napari/_qt/menus/file_menu.py

```diff
@@ -160,7 +162,31 @@ def _rebuild_samples_menu(self):
 
         self.open_sample_menu.clear()
 
-        for plugin_name, samples in plugin_manager._sample_data.items():
+        # eg ('dock', ('my_plugin', {'My widget': MyWidget}))
+        _iterable: Iterator[Tuple[str, Dict[str, SampleDict]]]
+        try:
+            import npe2
+        except ImportError:
+            _iterable = iter([])
+        else:
+            pm = npe2.PluginManager.instance()
+            _iterable = (
+                (
+                    plugin_name,
+                    {
+                        c.key: {
+                            'data': c.open,
+                            'display_name': c.display_name,
+                        }
+                        for c in contribs
+                    },
+                )
+                for plugin_name, contribs in pm.iter_sample_data()
```

Comment:
`plugin_name` is getting used below to create the menu for a multiprovider.  How do you feel about using the plugin display name instead?

Reply:
don't mind that, but maybe let's deal with that elsewhere?  The behavior here is consistent with npe1... it would be nice to keep the menus consistent somehow.